### PR TITLE
Use `table` printer in `show` commands

### PR DESCRIPTION
**Pull request by mkolesnik**
The code was needlessly complicated, this way it's simpler and has the same effect.

As part of the work:
    * Support for no max length was added.
    * Code now auto formats values (instead of having caller do it).
    * Spacing between columns is 3 (same as kubectl output)
    * Empty tables aren't printed (same as kubectl output)

The commands were greatly simplified by this change, and the print out
is much more aligned with `kubectl` behavior and with each other.

The PR encompasses 2 commits:
1. Overhaul `table.Printer` to be simple & useful
2. Use `table` printer in `show` commands

### CODE REVIEW

**Comment by mkolesnik** [comment]
Example outputs of `subctl show all`:

Before the change:

```shell
Cluster "cluster1"
 ✓ Detecting broker(s)
NAMESPACE                NAME                     COMPONENTS                        GLOBALNET ENABLED   GLOBALNET CIDR       DEFAULT GLOBALNET SIZE    DEFAULT DOMAINS                         
submariner-k8s-broker    submariner-broker        service-discovery, connectivity   true                242.0.0.0/8          65536                                                             

 ✓ Showing Connections
GATEWAY          CLUSTER   REMOTE IP   NAT  CABLE DRIVER  SUBNETS         STATUS     RTT avg.   
cluster2-worker  cluster2  172.19.0.8  no   libreswan     242.254.2.0/24  connected  194.57µs   

 ✓ Showing Endpoints
CLUSTER ID                    ENDPOINT IP     PUBLIC IP       CABLE DRIVER        TYPE            
cluster1                      172.19.0.6      176.230.47.24   libreswan           local           
cluster2                      172.19.0.8      176.230.47.24   libreswan           remote          

 ✓ Showing Gateways
NODE                            HA STATUS       SUMMARY                         
cluster1-worker                 active          All connections (1) are established

 ✓ Showing Network details
    Discovered network details via Submariner:
        Network plugin:  weave-net
        Service CIDRs:   [100.0.0.0/16]
        Cluster CIDRs:   [10.0.0.0/16]
        Global CIDR:     242.254.1.0/24

 ✓ Showing versions
COMPONENT                       REPOSITORY                                            VERSION         
submariner                      localhost:5000                                        local           
submariner-operator             localhost:5000                                        local           
service-discovery               localhost:5000                                        local           
```

After the change:

```shell
Cluster "cluster1"
 ✓ Detecting broker(s)
NAMESPACE               NAME                COMPONENTS                        GLOBALNET   GLOBALNET CIDR   DEFAULT GLOBALNET SIZE   DEFAULT DOMAINS   
submariner-k8s-broker   submariner-broker   service-discovery, connectivity   yes         242.0.0.0/8      65536                                      

 ✓ Showing Connections
GATEWAY           CLUSTER    REMOTE IP    NAT   CABLE DRIVER   SUBNETS          STATUS      RTT avg.     
cluster2-worker   cluster2   172.19.0.8   no    libreswan      242.254.2.0/24   connected   179.748µs    

 ✓ Showing Endpoints
CLUSTER    ENDPOINT IP   PUBLIC IP       CABLE DRIVER   TYPE     
cluster1   172.19.0.6    176.230.47.24   libreswan      local    
cluster2   172.19.0.8    176.230.47.24   libreswan      remote   

 ✓ Showing Gateways
NODE              HA STATUS   SUMMARY                               
cluster1-worker   active      All connections (1) are established   

 ✓ Showing Network details
    Discovered network details via Submariner:
        Network plugin:  weave-net
        Service CIDRs:   [100.0.0.0/16]
        Cluster CIDRs:   [10.0.0.0/16]
        Global CIDR:     242.254.1.0/24

 ✓ Showing versions
COMPONENT             REPOSITORY       VERSION   
submariner            localhost:5000   local     
submariner-operator   localhost:5000   local     
service-discovery     localhost:5000   local     
```

File: internal/show/connections.go [hunk 1]
```diff
@@ -56,38 +43,44 @@ func Connections(clusterInfo *cluster.Info, status reporter.Interface) bool {
...
+	printer := table.Printer{Columns: []table.Column{
+		{Name: "GATEWAY", MaxLength: 30},
+		{Name: "CLUSTER", MaxLength: 24},
+		{Name: "REMOTE IP"},
+		{Name: "NAT"},
+		{Name: "CABLE DRIVER"},
+		{Name: "SUBNETS", MaxLength: 40},
+		{Name: "STATUS"},
+		{Name: "RTT avg."},
+	}}
```
Comment (skitt) [inline internal/show/connections.go:55]: All the changes to this file should be in the second commit.

File: internal/show/table/table.go [hunk 2]
```diff
@@ -18,73 +18,116 @@ limitations under the License.
...
-	for _, header := range p.Headers {
-		headerRow = append(headerRow, header.Name)
+func printRow(template string, row []string) {
+	values := make([]interface{}, len(row))
+	for i, v := range row {
+		values[i] = v
 	}
 
-	rowList := [][]string{headerRow}
+	fmt.Printf(template, values...)
```
Comment (skitt) [inline internal/show/table/table.go:90]: Doesn’t

    fmt.Printf(template, row...)

work here?

**Review by skitt — CHANGES_REQUESTED** [review]

**Comment by skitt** [comment]
That was quick, thanks for taking care of this!

File: internal/show/connections.go (hunk 1 above)
Comment (mkolesnik) [reply internal/show/connections.go:55]: Can't because `connections` is already using the printer

File: internal/show/table/table.go (hunk 2 above)
Comment (mkolesnik) [reply internal/show/table/table.go:90]: I wish...

```
cannot use row (variable of type []string) as []interface{} value in argument to fmt.Printf
```

File: internal/show/connections.go (hunk 1 above)
Comment (skitt) [reply internal/show/connections.go:55]: Oh yes, I can’t read

File: internal/show/table/table.go (hunk 2 above)
Comment (skitt) [reply internal/show/table/table.go:90]: Indeed...

**Review by skitt — APPROVED** [review]

**Review by dfarrell07 — APPROVED** [review]
Much simpler and less bug-prone, thanks @mkolesnik!

**Pull request merged**